Formula/step.rb

```diff
@@ -0,0 +1,114 @@
+      bash_completion.install "autocomplete/bash_autocomplete" => "step"
+    end
+
+    resource("certificates").stage do
+      contents = Dir["{*,.git,.gitignore}"]
+      (buildpath/"src/github.com/smallstep/certificates").install contents
+      cd "#{buildpath}/src/github.com/smallstep/certificates" do
+        system "make", "build"
+        bin.install "bin/step-ca" => "step-ca"
+      end
+    end
+  end
+
+  test do
+    # Generate a public / private key pair. Creates foo.pub and foo.priv.
+    system "#{bin}/step", "crypto", "keypair", "foo.pub", "foo.priv", "--no-password", "--insecure"
+    assert_predicate testpath/"foo.pub", :exist?
+    assert_predicate testpath/"foo.priv", :exist?
+
+    # Generate a root certificate and private key with subject baz written to baz.crt and baz.key.
+    system "#{bin}/step", "certificate", "create", "--profile", "root-ca",
+        "--no-password", "--insecure", "baz", "baz.crt", "baz.key"
+    assert_predicate testpath/"baz.crt", :exist?
+    assert_predicate testpath/"baz.key", :exist?
+    baz_crt = File.read(testpath/"baz.crt")
+    assert_match(/^-----BEGIN CERTIFICATE-----.*/, baz_crt)
+    assert_match(/.*-----END CERTIFICATE-----$/, baz_crt)
+    baz_key = File.read(testpath/"baz.key")
+    assert_match(/^-----BEGIN EC PRIVATE KEY-----.*/, baz_key)
+    assert_match(/.*-----END EC PRIVATE KEY-----$/, baz_key)
+    shell_output("#{bin}/step certificate inspect --format json baz.crt > baz_crt.json")
+    baz_crt_json = JSON.parse(File.read(testpath/"baz_crt.json"))
+    assert_equal "CN=baz", baz_crt_json["subject_dn"]
+    assert_equal "CN=baz", baz_crt_json["issuer_dn"]
+
+    # Generate a leaf certificate signed by the previously created root.
+    system "#{bin}/step", "certificate", "create", "--profile", "intermediate-ca",
+        "--no-password", "--insecure", "--ca", "baz.crt", "--ca-key", "baz.key",
+        "zap", "zap.crt", "zap.key"
+    assert_predicate testpath/"zap.crt", :exist?
+    assert_predicate testpath/"zap.key", :exist?
+    zap_crt = File.read(testpath/"zap.crt")
+    assert_match(/^-----BEGIN CERTIFICATE-----.*/, zap_crt)
+    assert_match(/.*-----END CERTIFICATE-----$/, zap_crt)
+    zap_key = File.read(testpath/"zap.key")
+    assert_match(/^-----BEGIN EC PRIVATE KEY-----.*/, zap_key)
+    assert_match(/.*-----END EC PRIVATE KEY-----$/, zap_key)
+    shell_output("#{bin}/step certificate inspect --format json zap.crt > zap_crt.json")
+    zap_crt_json = JSON.parse(File.read(testpath/"zap_crt.json"))
+    assert_equal "CN=zap", zap_crt_json["subject_dn"]
+    assert_equal "CN=baz", zap_crt_json["issuer_dn"]
+
+    # Initialize a PKI and step-ca configuration, boot the CA, and create a
+    # certificate using the API.
+    (testpath/"password.txt").write("password")
+    steppath = "#{testpath}/.step"
+    Dir.mkdir(steppath) unless File.exist?(steppath)
+    ENV["STEPPATH"] = steppath
+    system "#{bin}/step", "ca", "init", "--address", "127.0.0.1:8081",
```

Comment:
This line [is failing on Jenkins](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/37823/version=high_sierra/testReport/junit/brew-test-bot/high_sierra/test_step/) with the message:

```
==> /usr/local/Cellar/step/0.8.4/bin/step ca init --address 127.0.0.1:8081 --dns 127.0.0.1 --password-file /private/tmp/step-test-20190212-3676-p935hj/password.txt --provisioner-password-file /private/tmp/step-test-20190212-3676-p935hj/password.txt --name homebrew-smallstep-test --provisioner brew
error allocating terminal: open /dev/tty: device not configured
Error: step: failed
Failed executing:
/usr/local/Homebrew/Library/Homebrew/formula.rb:1921:in `block in system'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1858:in `open'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1858:in `system'
```

Reply:
Yep, we're in the process of fixing that right now. Looks like we were opening up prompt terminals even when the prompt was satisfied via the cli. I should be pushing an update in 15-30 mins.